joss/paper.md

```diff
@@ -0,0 +1,55 @@
+
+# Statement of Need
+
+While multiple computational evolution software packages include system-specific phylogeny tracking [ofria2004avida;@bohm2017mabe;@de2012deap;@Garwood_REvoSim_Organism-level_simulation_2019], to our knowledge no other general-purpose perfect phylogeny tracking library exists. The two most closely related tools are heriditary stratigraphy (hstrat) [@moreno2022hereditary] and Automated Phylogeny Over Geological Timescales (APOGeT) [@godin2019apoget]. Both of these tools are designed to plug in to computational evolution software and record phylogenetic information, however they both fill slightly more advanced roles than Phylotrackpy. Hstrat sacrifices maintaining perfect information to improve scalability, while the emphasis of APOGeT is on tracking speciation in sexually-reproducing populations. In contrast, the role of Phylotrackpy is simply to perfectly record asexual phylogenies.
+
+It is important to note that the problem of tracking a phylogeny in real time is different than the problem of phylogenetic reconstruction traditionally faced by biologists. Essentially all bioinformatics tools related to phylogenies are focused on inferring phylogenies from extant organisms (and sometimes fossils) and working with these reconstructed phylogenies. Recorded phylogenies have numerous subtle differences from reconstructed phylogenies, outlined in the Phylotrackpy documentation.
+
+Phylotrackpy has already facilitated a variety of research projects. Originally developed as part of the Empirical library [@ofria2020empirical], the C++ backend has been integrated into packages such as Modular Agent-Based Evolver (MABE) 2.0 [@bohm2019mabe], Symbulation [@vostinarSpatialStructureCan2019;@johnsonEndosymbiosisBustInfluence2022a], and even a fork of Avida [@ofria2004avida;@dolsonInterpretingTapeLife2020]. It has facilitated research on open-ended evolution [@dolsonMODESToolboxMeasurements2019], the origin of endosymbiosis [@johnsonEndosymbiosisBustInfluence2022a], the importance of phylogenetic diversity for machine learning via evolutionary computation [@hernandez2022can;@shahbandegan2022untangling], and more. The Python wrapper for Phylotrackpy is newer, but it has already served as a point of comparison in the development of other phylogenetic tools [@moreno2022hereditary].
```

Comment:
I suspect some packages probably implicitly do something tracking-like because there's a bunch of work looking at the quality of reconstructions based on simulated sequence data? Working on how to phrase this.

Reply:
ah but it's not agent based. Can still include simulation of phylogenies on the laundry list.